### DISCUSSION TITLE
Is there a way to know when token allow spending or confirm transaction or sign transaction is displayed

### DISCUSSION THREAD

**Discussion by git-josip** [Synthetixio/synpress#780] [Q&A]
I have issues that sometimes I have a token spending screen and sometimes is not there. 
If I put the command cy.confirmTransaction() it will fail if there is approve token spending and also if there is not approveTokenSpending it will fail if I try to approve it. 

How to approach this?

**Comment by meanMonk** [comment]
Is there any solution this? I am also facing the same issue. 

I have allowance approval popup for my erc20 token once confirm we trigger the transfer so can we deal with two modal from meta mask. 

1. allowance approval modal.
2. confirm transfer of token modal.


I have tried with below approach but only work for any one modal.
```typescript.
await metamask.confirmPermissionToSpend(amount, true)

````
